native/lambda_game_engine/src/effect.rs

```diff
@@ -8,6 +8,7 @@ pub struct Effect {
     pub effect_time_type: TimeType,
     pub player_attributes: Vec<AttributeChange>,
     pub projectile_attributes: Vec<AttributeChange>,
+    pub skills_to_execute: Vec<String>,
```

Comment:
This is a complicated one, doing this causes a circular dependency between `Effect` and `SkillConfig`

This was something that was heavily avoided to prevent invalid configs (e.g. skill uses effect that does not exist)

I'm not opposed to this because the idea makes sense, but I think we should rethink a bit on the config parsing to prevent this `String` usage so there are no dynamic finds at runtime

Reply:
Discussed this offline, this `String` is actually the skill_key used to identify the skill when activating, not the `Skill` unique name like in other part of the config so it was decided that all good due to being the same behaviour as activating a skill